src/affine.py

```diff
@@ -582,41 +583,12 @@ def __mul__(self, other):
             return NotImplemented
 
     def __rmul__(self, other):
-        """Right hand multiplication.
-
-        .. deprecated:: 2.3.0
-            Right multiplication will be prohibited in version 3.0. This method
-            will raise AffineError.
-
-        Parameters
-        ----------
-        other : Affine or iterable of (vx, vy)
-
-        Returns
-        -------
-        tuple of two floats
-
-        Notes
-        -----
-        We should not be called if other is an affine instance This is
-        just a guarantee, since we would potentially return the wrong
-        answer in that case.
-        """
-        warnings.warn(
-            "Right multiplication will be prohibited in version 3.0",
-            DeprecationWarning,
-            stacklevel=2,
-        )
-        assert not isinstance(other, Affine)
-        return self.__mul__(other)
+        return NotImplemented
```

Comment:
Also not related to type hinting: removal of the right multiplication implementation.

Reply:
Also note: I'm not providing type hints for dunder methods. These are specified by the Python language.